asllib/doc/ASLTypingReference.tex

```diff
@@ -495,149 +681,276 @@ \section{Domain of a Type}
 The definition of a dynamic domain refers to \emph{dynamic environments}, denoted $\dynamicenvs$,
 which assigns native values to identifiers~\cite{ASLSemanticsReference}.
 
+\hypertarget{def-envs}{}
+We define \emph{environments} as pairs of static environments and dynamic environments:
+$\envs = \staticenvs \times \dynamicenvs$.
+
 Formally, the partial function
 \[
-  \dynamicdomain : \overname{\staticenvs}{\tenv} \times \overname{\dynamicenvs}{\denv} \times \overname{\ty}{\vt}
-  \partialto \pow{\vals}
+  \dynamicdomain : \overname{\envs}{\env} \times \overname{\ty}{\vt}
+  \partialto \overname{\pow{\vals}}{\vd}
 \]
-assigns the set of values that a type $\vt$ can hold, given the static environment $\tenv$ and dynamic environment $\denv$.
+assigns the set of values that a type $\vt$ can hold in a given environment $\env$.
 %
-We say that $\dynamicdomain(\tenv, \denv, \vt)$ is the \emph{dynamic domain} of $\vt$
-(in the given static environment $\tenv$ and dynamic environment $\denv$).
+We say that $\dynamicdomain(\env, \vt)$ is the \emph{dynamic domain} of $\vt$
+in the environment $\env$.
 %
 The \emph{static domain} of a type is the set of values which storage elements of that type may hold
 \underline{across all possible dynamic environments}.
 %
 The reason that for this distinction is that the sets of values
 considered for bitvector types, array types, constrained integers, and constrained bitvectors,
-can be parameterized by configuration variables and subprogram parameters whose values are determined dynamically.
+can depend on the dynamic values of variables.
 
-Types that do not refer to subprogram parameters have a static domain that is equal to their dynamic domain.
+Types that do not refer to variables whose values are only known dynamically have
+a static domain that is equal to any of their dynamic domains.
 In those cases, we simply refer to their \emph{domain}.
 
-Associating a set of values to a type is done by first substituting its configuration variables
-and subprogram parameters by their values in the dynamic environment. This is done via the
-partial function $\substparams$ (see \secref{TypingRule.SubstituteParameterValues}).
+Associating a set of values to a type is done by evaluating any expressions appearing
+in the type definitions via the ASL semantics~\cite{ASLSemanticsReference},
```

Comment:
I think this needs a little more hand-holding - maybe split into two as follows:
... is done by evaluating any expression appearing in the type definitions. Evaluation is done using the ASL semantics~\cite{...}, more precisely using the relation ...